smarts/env/tests/test_determinism.py

```diff
@@ -138,7 +138,7 @@ def check_callback(rewards, agent_obs, dones, index):
             agent_obs.waypoint_paths, orig_agent_obs.waypoint_paths
         ):
             for now, prev in zip(w_paths_now, w_paths_prev):
-                assert now.id == prev.id
+                assert hash(now) == hash(prev)
```

Comment:
Implement `__eq__` on Waypoint instead, that way we can simply compare waypoints i.e. `now == prev` and you can remove all the remaining equality checking code below since it's just checking what `__eq__` will check.